torchdata/datapipes/iter/load/s3io.py

```diff
@@ -0,0 +1,87 @@
+from io import BytesIO
+from typing import Iterator, Tuple
+
+import torchdata
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+
+@functional_datapipe("list_file_by_s3")
+class S3FileListerIterDataPipe(IterDataPipe[str]):
+    r""":class:`S3FileListerIterDataPipe`.
+
+    Iterable DataPipe that lists URLs with the given prefixes.
+
+    Args:
+        source_datapipe: a DataPipe that contains URLs/URL prefixes to s3 files
+        length: Nominal length of the datapipe
+        requestTimeoutMs: optional, overwrite the default timeout setting for this datapipe
+        region: optional, overwrite the default region inferred from credentials for this datapipe
+
+    Note:
+        AWS_CPP_SDK is necessary to use the S3 DataPipe(s).
+    """
```

Comment:
```suggestion
    r"""
    Iterable DataPipe that lists URLs with the given prefixes (functional name: ``list_file_by_s3``).

    Args:
        source_datapipe: a DataPipe that contains URLs/URL prefixes to s3 files
        length: Nominal length of the datapipe
        requestTimeoutMs: optional, overwrite the default timeout setting for this datapipe
        region: optional, overwrite the default region inferred from credentials for this datapipe

    Note:
        AWS_CPP_SDK is necessary to use the S3 DataPipe(s).
     
    Example:
        >>> # Add usage example here
    """
```

Minor comment on docstring to ensure the rendered documentation will look consistent.

Reply:
Updated docstring for both functions with functional names, and updated examples